File: 1415. The k-th Lexicographical String of All Happy Strings of Length n.cpp
class Solution {
    public:
        string s = "abc", ans;
        void backtrack(string cur, int &n, int &k, int &count)
        {
            if(count == k) return;
            if(cur.size() == n)
            {
                ans = cur, count += 1;
                return;
            }
            for(auto ch : s)
            {
                if(ch == cur.back()) continue;
                backtrack(cur + ch, n, k, count);
            }
        }
        string getHappyString(int n, int k) 
        {
            int count = 0;
            for(auto ch : s)
            {
                string cur;
                cur += ch;
                backtrack(cur, n, k, count);
            }
            return count == k ? ans : "";
        }
    };

File: Backtracking/1415. The k-th Lexicographical String of All Happy Strings of Length n.py
class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        store = []
        def dfs(s):
            if len(s) > 1:
                if s[-1] == s[-2]:
                    return 
            if len(s) == n:
                store.append(s)
                return 
            dfs(s + 'a')
            dfs(s + 'b')
            dfs(s + 'c')
        dfs("")

        store.sort()
        if k <= len(store):
            return store[k - 1]
        return ""